Tests/webcomponents/js/ez-universal-discovery.js

```diff
@@ -0,0 +1,409 @@
+        });
+
+        it('should have received a cancelDiscoverHandler function', function () {
+            const view = getViewMock(container);
+
+            renderViewStub = sinon.stub(window.eZ.YUI.app, 'renderSideView', function (View, Service, params, done) {
+                assert.isFunction(params.cancelDiscoverHandler);
+                done(false, 'whatever', view);
+            });
+            document.dispatchEvent(new CustomEvent('ez:yui-app:ready'));
+        });
+
+        it('should have received a contentDiscoveredHandler function', function () {
+            const view = getViewMock(container);
+
+            renderViewStub = sinon.stub(window.eZ.YUI.app, 'renderSideView', function (View, Service, params, done) {
+                assert.isFunction(params.contentDiscoveredHandler);
+                done(false, 'whatever', view);
+            });
+            document.dispatchEvent(new CustomEvent('ez:yui-app:ready'));
+        });
+
+        it('should have received a isSelectable function', function () {
+            const view = getViewMock(container);
+
+            renderViewStub = sinon.stub(window.eZ.YUI.app, 'renderSideView', function (View, Service, params, done) {
+                assert.isFunction(params.isSelectable);
+                done(false, 'whatever', view);
+            });
+            document.dispatchEvent(new CustomEvent('ez:yui-app:ready'));
+        });
+
+        it('should handle loading error', function () {
+            const errorMsg = 'error message';
+
+            renderViewStub = sinon.stub(window.eZ.YUI.app, 'renderSideView', function (View, Service, params, done) {
+                done(new Error(errorMsg));
+            });
+            document.dispatchEvent(new CustomEvent('ez:yui-app:ready'));
+
+            assert.equal(
+                universalDiscovery.innerHTML,
+                errorMsg,
+                '<ez-universal-discovery> should contain the error message'
+            );
+        });
+
+        it('should add the ez-js-standard-form class', function () {
+            const view = getViewMock(container);
+
+            renderViewStub = sinon.stub(window.eZ.YUI.app, 'renderSideView', function (View, Service, params, done) {
+                done(false, 'whatever', view);
+            });
+            document.dispatchEvent(new CustomEvent('ez:yui-app:ready'));
+            assert.isTrue(universalDiscovery.classList.contains('ez-js-standard-form'));
+        });
+
+        describe('events', function () {
+
+            describe('`ez:cancel`', function () {
+                it('should be dispatched', function() {
+                    let isDispatched = false;
+
+                    universalDiscovery.addEventListener('ez:cancel', function (e) {
+                        assert.isFalse(e.bubbles);
+                        isDispatched = true;
+                    });
+                    dispatchEvent('cancelDiscoverHandler', {});
```

Comment:
the cancel discover handler does not accept any selection so the `{}` is a bit confusing, you should just omit this parameter